### PR TITLE
Antagus Minor:Spritus

### PR DESCRIPTION
**Pull request by Aurorablade**
Most of these are updates from TG...half of them i cannot find in a history search of PRS....but i know this one https://github.com/tgstation/-tg-station/pull/16110 <3 you at TG!
Updates guardians:
New icons for magic, some new colors for holoparasite...Bio is still using the headcrab sprite until someone makes me a new one.
New naming for magic and holo.
Ranged Night vision
New attack fo choas type, non teleport. (i missed adding some of this code, will work on it later..shhhhhhhhhhhhhhhh)

I NEED to figure out how to add a burst fire to ranged to buff them up a bit.

:cl: Fethas
rscadd:New sprites,and names for magic/holoparasite
rscadd:Guardians now see a hud display of summoner health
rscadd: Chaos types no longer random teleport, but will make the target hallucinate everyone looks like the guardian.
/:cl:

### CODE REVIEW

**Comment by Aurorablade** [comment]
guardian - copy
Icon preview

**Comment by FalseIncarnate** [comment]
Might want to update your OP to include the rapid fire for ranged guardians.

Also, is it possible to give the Chaos Guardian a toggle between teleport and hallucination? Would give them some nice versatility for sowing chaos in the crowds.

**Comment by ScratKnapp** [comment]
This must be the work of an enemy stand!

:100: 

Also, yes please burst attack for Ranged. Their ranged attack is pretty shit.

**Comment by Fox-McCloud** [comment]
Oh. Very nice; I like that we can do custom hallucinations, now.

**Comment by Aurorablade** [comment]
@FalseIncarnate I would need to figure out how to do an attack toggle properly.....

**Comment by FalseIncarnate** [comment]
I'll work with you tomorrow to try and tackle that. Got a basic idea in mind, but I'm sure the code will be uncooperative as usual

File: code/modules/flufftext/Hallucination.dm [hunk 1]
```diff
@@ -325,29 +325,33 @@ Gunshots/explosions/opening doors/less rare audio (done)
 /obj/effect/hallucination/delusion
 	var/list/image/delusions = list()
 
-/obj/effect/hallucination/delusion/New(loc,var/mob/living/carbon/T)
+/obj/effect/hallucination/delusion/New(loc,mob/living/carbon/T,force_kind = null , duration = 300,skip_nearby = 1, custom_icon = null, custom_icon_file = null)
```
Comment (tigercat2000) [inline code/modules/flufftext/Hallucination.dm]: the spacing in these arguments hurts me physically

File: code/modules/flufftext/Hallucination.dm [hunk 2]
```diff
@@ -325,29 +325,33 @@ Gunshots/explosions/opening doors/less rare audio (done)
...
 	target = T
 	var/image/A = null
-	var/kind = rand(1,3)
+	var/kind = force_kind ? force_kind : pick("clown","corgi","carp","skeleton","demon")
```
Comment (tigercat2000) [inline code/modules/flufftext/Hallucination.dm]: spacing in pick

File: code/game/gamemodes/miniantags/guardian/guardian.dm [hunk 3]
```diff
@@ -233,13 +248,29 @@
...
-	if(prob(45))
-		if(istype(target, /atom/movable))
-			var/atom/movable/M = target
-			if(!M.anchored && M != summoner)
-				do_teleport(M, M, 10)
+	if(..())
+		if(toggle)
+			if(ishuman(target) && !summoner)
+				spawn(0)
+					new /obj/effect/hallucination/delusion(target.loc,target,force_kind="custom",duration=200,skip_nearby=0, custom_icon = src.icon_state, custom_icon_file = src.icon)
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/guardian/guardian.dm]: this spacing tho

File: code/game/gamemodes/miniantags/guardian/guardian.dm [hunk 4]
```diff
@@ -118,20 +134,16 @@
 		return
 	if(!summoner) return
 	if(loc == summoner)
-		loc = get_turf(summoner)
+		forceMove(summoner)
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/guardian/guardian.dm]: forceMove(get_turf(summoner))

File: code/game/gamemodes/miniantags/guardian/guardian.dm [hunk 5]
```diff
@@ -80,6 +84,17 @@
 	summoner << "<span class='danger'><B>Your [name] died somehow!</span></B>"
 	summoner.death()
 
+
+/mob/living/simple_animal/hostile/guardian/proc/updatehudhealth()
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/guardian/guardian.dm]: use `/mob/living/simple_animal/hostile/guardian/handle_hud_icons_health()` and remove the Life() hook

File: code/game/gamemodes/miniantags/guardian/guardian.dm [hunk 6]
```diff
@@ -60,7 +60,11 @@
 			src << "You moved out of range, and were pulled back! You can only move [range] meters from [summoner.real_name]"
 			visible_message("<span class='danger'>The [src] jumps back to its user.</span>")
 			Recall()
-	if(summoned && !summoner)
+	if(summoned && !summoner && !adminseal)
+		src << "<span class='danger'>You somehow lack a summoner! As a result, you dispel!</span>"
+		ghostize()
+		qdel()
+	if(!summoner && !adminseal)
 		src << "<span class='danger'>You somehow lack a summoner! As a result, you dispel!</span>"
```
Comment (Crazylemon64) [inline code/game/gamemodes/miniantags/guardian/guardian.dm]: Won't this mean that admin-spawned guardians will automatically dispel?

File: code/modules/flufftext/Hallucination.dm (hunk 1 above)
Comment (TheDZD) [inline code/modules/flufftext/Hallucination.dm]: This one makes me scream too because it's inconsistent.

Comment (Aurorablade) [inline code/modules/flufftext/Hallucination.dm]: ....i though i fixed that one..

Comment (TheDZD) [inline code/modules/flufftext/Hallucination.dm]: You fixed one of them, but there's still a missing space after `loc`.

**Comment by ScratKnapp** [comment]
Would it be possible to throw in a fix for #3814 before this is merged?

**Comment by Fox-McCloud** [comment]
May want to credit TG's PR for some of these updates, especially since one is active and unmerged, as of yet.

**Comment by Fox-McCloud** [comment]
I'm really iffy about them having the ability to do both, but we'll see.

Guardians are, by far, the most powerful thing on the entire uplink, especially since they carry no risk of failure on our server.

**Pull request merged**